app/javascript/packages/document-capture/components/in-person-location-full-address-entry-post-office-search-step.tsx

```diff
@@ -1,14 +1,136 @@
+import { useState, useEffect, useCallback, useRef, useContext } from 'react';
 import { useI18n } from '@18f/identity-react-i18n';
-import { PageHeading } from '@18f/identity-components';
+import { Alert, PageHeading } from '@18f/identity-components';
+import { request } from '@18f/identity-request';
+import { forceRedirect } from '@18f/identity-url';
+import {
+  transformKeys,
+  snakeCase,
+  LocationQuery,
+  LOCATIONS_URL,
+} from '@18f/identity-address-search';
+import FullAddressSearch from './in-person-full-address-search';
 import BackButton from './back-button';
+import AnalyticsContext from '../context/analytics';
+import InPersonLocations, { FormattedLocation } from './in-person-locations';
+import { InPersonContext } from '../context';
+import UploadContext from '../context/upload';
 
-function InPersonLocationFullAddressEntryPostOfficeSearchStep({ toPreviousStep }) {
+function InPersonLocationFullAddressEntryPostOfficeSearchStep({
+  onChange,
+  toPreviousStep,
+  registerField,
+}) {
+  const { inPersonURL } = useContext(InPersonContext);
   const { t } = useI18n();
+  const [inProgress, setInProgress] = useState<boolean>(false);
+  const [isLoadingLocations, setLoadingLocations] = useState<boolean>(false);
+  const [autoSubmit, setAutoSubmit] = useState<boolean>(false);
+  const { trackEvent } = useContext(AnalyticsContext);
+  const [locationResults, setLocationResults] = useState<FormattedLocation[] | null | undefined>(
+    null,
+  );
+  const [foundAddress, setFoundAddress] = useState<LocationQuery | null>(null);
+  const [apiError, setApiError] = useState<Error | null>(null);
+  const [disabledAddressSearch, setDisabledAddressSearch] = useState<boolean>(false);
+  const { flowPath } = useContext(UploadContext);
+
+  const onFoundLocations = (
+    address: LocationQuery | null,
+    locations: FormattedLocation[] | null | undefined,
+  ) => {
+    setFoundAddress(address);
+    setLocationResults(locations);
+  };
+
+  // ref allows us to avoid a memory leak
+  const mountedRef = useRef(false);
+
+  useEffect(() => {
+    mountedRef.current = true;
+    return () => {
+      mountedRef.current = false;
+    };
+  }, []);
+
+  // useCallBack here prevents unnecessary rerenders due to changing function identity
+  const handleLocationSelect = useCallback(
+    async (e: any, id: number) => {
+      if (flowPath !== 'hybrid') {
+        e.preventDefault();
+      }
+      const selectedLocation = locationResults![id]!;
```

Comment:
what does this signify `locationResults![id]!`, i haven't seen this syntax before